### PR TITLE
Increase number of dependencies in v5e maxtext perf tests

### PR DESCRIPTION
**Pull request by raymondzouu**
# Description

Increase number of dependencies in v5e maxtext perf tests due to workload start timeouts.

DAG dependencies now look like this https://screenshot.googleplex.com/7TfW2uPwC3xo9Rn

# Tests

N/A

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: dags/multipod/maxtext_v5e_configs_perf.py [hunk 1]
```diff
@@ -75,11 +75,9 @@
...
-        (
-            maxtext_sweep_gke_test[i].run_with_run_name_generation()
-            >> maxtext_sweep_gke_test[
-                num_tests - 1 - i
-            ].run_with_run_name_generation()
-        )
+      prev = maxtext_sweep_gke_test[0].run_with_run_name_generation()
+      for i in range(1, len(maxtext_sweep_gke_test)):
+        curr = maxtext_sweep_gke_test[i].run_with_run_name_generation()
+        if i % 4 != 0:
```
Comment (RissyRan) [inline dags/multipod/maxtext_v5e_configs_perf.py:81]: Nit: let's move this 4 to a variable? i.e. something like `chain_num=4`?

**Review by RissyRan — APPROVED** [review]

File: dags/multipod/maxtext_v5e_configs_perf.py (hunk 1 above)
Comment (raymondzouu) [reply dags/multipod/maxtext_v5e_configs_perf.py]: Done!

File: dags/multipod/maxtext_v5e_configs_perf.py [hunk 2]
```diff
@@ -75,11 +75,10 @@
...
-                num_tests - 1 - i
-            ].run_with_run_name_generation()
-        )
+      chain_num = 4
+      prev = maxtext_sweep_gke_test[0].run_with_run_name_generation()
+      for i in range(1, len(maxtext_sweep_gke_test)):
+        curr = maxtext_sweep_gke_test[i].run_with_run_name_generation()
+        if i % chain_num != 0:
+          prev >> curr
+        prev = curr
```
Comment (jonb377) [inline dags/multipod/maxtext_v5e_configs_perf.py:84]: Not blocking, but I've been interested in trying out [chain](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/models/baseoperator/index.html#airflow.models.baseoperator.chain), it takes some number of tasks and adds dependencies within the groups (e.g. `chain([a, b], [c, d])` will chain a->b and c->d). The tricky part would be finding a clean way to break up the original list of tasks.

If you want to see how it looks, you can try something like this, but it's harder to read so I'd prefer to keep what you have:

```python
from airflow.models.baseoperator import chain
chain(*(maxtext_sweep_gke_test[n:n+chain_num] for n in range(0, len(maxtext_sweep_gke_test), chain_num)))
```

**Review by jonb377 — APPROVED** [review]
Thanks Raymond!

File: dags/multipod/maxtext_v5e_configs_perf.py (hunk 2 above)
Comment (raymondzouu) [reply dags/multipod/maxtext_v5e_configs_perf.py:84]: Thanks for the suggestion! I'll give that a try in the future and keep it as is for now

**Pull request merged**